### PR TITLE
Create an HOC EnhanceWithCopy

### PR DESCRIPTION
**Pull request by Chochek**

### CODE REVIEW

**Comment by Chochek** [comment]
[(2) Reactions: Create an HOC “EnhanceWithCopy”](https://trello.com/c/PbUB3HuS/1278-2-reactions-create-an-hoc-enhancewithcopy)

File: src/components/EnhanceWithCopy/EnhanceWithCopy.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import { string } from 'prop-types'
+import Clipboard from 'clipboard'
+import Tooltip from '../Tooltip'
+
+const EnhanceWithCopy = WrappedComponent => {
+  return class extends React.Component {
+    static propTypes = {
+      id: string.isRequired,
+      tooltipPlacement: string
```
Comment (samageloff) [inline src/components/EnhanceWithCopy/EnhanceWithCopy.js]: We need the `getValue` method added to props

File: src/components/EnhanceWithCopy/EnhanceWithCopy.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      // Set the copied state to true
+      this.setState({ copied: true }, () => {
+        // Focus the trigger element if it needs to be focused (input or textarea)
+        this.focusTriggerElement()
+
+        setTimeout(() => {
+          // Set the copied state to false
+          this.setState({ copied: false }, () => {
+            // Focus the trigger element if it needs to be focused (input or textarea)
+            this.focusTriggerElement()
```
Comment (samageloff) [inline src/components/EnhanceWithCopy/EnhanceWithCopy.js]: Why do we need to call this again?

Comment (Chochek) [reply src/components/EnhanceWithCopy/EnhanceWithCopy.js]: Every time the state is set the input field gets blurred

**Pull request merged**